pkg/controller/webhook/util.go

```diff
@@ -56,43 +56,52 @@ func NewValidatingResource(resourcePluralName string) schema.GroupVersionResourc
 
 // Allowed returns true if the admission request for the plural name of the
 // resource passed in should be allowed to pass through, false otherwise.
-func Allowed(a admission.Request, pluralResourceName string, status *admission.Response) bool {
+func Allowed(a admission.Request, pluralResourceName string) *admission.Response {
 	// We want to let through:
 	// - Requests that are not for create, update
 	// - Requests for things that are not <pluralResourceName>
 	createOrUpdate := a.Operation == admissionv1beta1.Create || a.Operation == admissionv1beta1.Update
 	isMyGroupAndResource := a.Resource.Group == v1beta1.SchemeGroupVersion.Group && a.Resource.Resource == pluralResourceName
 	if !createOrUpdate || !isMyGroupAndResource {
-		status.Allowed = true
-		return true
+		return &admission.Response{
+			AdmissionResponse: admissionv1beta1.AdmissionResponse{
+				Allowed: true,
+			},
+		}
 	}
-	return false
+	return nil
```

Comment:
I guess return `nil` equals to `not allowed` would bring ambiguous, and return a `admission.Response` will break the function as well.

The function name `Allowed` designed to check whether a specific request is allowed. So the return type should be `bool`.

A simple way maybe just remove `status` from the original function signature.

Reply:
Thanks for the feedback, @RainbowMango, I'll look to improve the signature of this func.

Reply:
@RainbowMango I addressed your concern and improved the utility functions a bit. PTAL.